backstory/src/main/java/com/google/sps/story/DatamuseRequestClient.java

```diff
@@ -0,0 +1,150 @@
+    this(DATAMUSE_URL);
+  }
+
+  /**
+   * Constructs a request client that will query
+   * the given URL (for testing purposes)
+   *
+   * @param queryUrl the url to instantiate this request client with
+   */
+  public DatamuseRequestClient(String queryUrl) {
+    this.queryUrl = queryUrl;
+  }
+
+  /**
+   * Gets an array of adjectives related to the passed-in noun, which will have
+   * a passed-in cap on its size (array returned might be smaller if database
+   * does not have enough related adjectives). Fetches these adjectives
+   * by querying the Datamuse database.
+   *
+   * @param noun the noun to get adjectives related to (must be one word)
+   * @param cap the maximum number of adjectives to retrieve
+   * @return an array of adjectives related to the noun
+   * @throws IllegalArgumentException if noun is more than one word (has a space in it)
+   * @throws APINotAvailableException if Datamuse API cannot be reached
+   * @throws RuntimeException if JSON received back from the API cannot be parsed (JSONException)
+   *    or does not have objects of type JSON (ClassCastException)
+   */
+  public String[] getRelatedAdjectives(String noun, int cap)
+      throws IllegalArgumentException, APINotAvailableException, RuntimeException {
+    if (noun.contains(" ")) {
+      throw new IllegalArgumentException("Noun must be one word.");
+    }
+
+    // put together a query asking for adjectives related to the noun (rel_jjb=noun)
+    // and capping the number of results at 10 (max=10)
+    String query = queryUrl + "rel_jjb=" + noun + "&max=" + cap;
+    String jsonResponse = querySite(query);
+
+    // there are two try statements b/c there's a possibility for more than one JSONException to be
+    // thrown in this method so specifying what went wrong with that exact JSON exception
+    // will be more helpful hence a try block for each JSONException that needs a customized message
+    JSONArray jsonArray;
+
+    try {
+      jsonArray = new JSONArray(jsonResponse);
+    } catch (JSONException exception) {
+      throw new RuntimeException(
+          "Could not parse the JSON received back from the Datamuse Query.", exception);
+    }
+
+    try {
+      int length = jsonArray.length();
+      String[] adjectives = new String[length];
+
+      for (int i = 0; i < length; i++) {
+        JSONObject jsonObject = (JSONObject) jsonArray.get(i);
+        adjectives[i] = jsonObject.getString("word");
+      }
+
+      return adjectives;
+    } catch (ClassCastException exception) {
+      throw new RuntimeException("JSON array received was not of JSON objects.", exception);
+    } catch (JSONException exception) {
+      throw new RuntimeException(
+          "JSON Object did not have a String value for the key \"word\".", exception);
+    }
+  }
+
+  /**
+   * Query the site at the given url and retrieve the content (likely JSON) from that
+   * site. If there's nothing at that site, return null. This code was
```

Comment:
Suggest removing "site" from the docs, as I think that's often used to represent something more than a HTTP resource.

I'd also recommend stating what type of request is being made, e.g. `Makes a GET request to the provided url.`

Reply:
That makes sense.

Reply:
So I'm guessing that this UrlConnection is a HTTPUrlConnection, and that by default, the method is `GET`:
https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#method

i.e. the UrlConnection object is making a GET request to that URL, and then exposing the response as a stream.

You could add logging to confirm.

Reply:
In that case, I'm going to add that it was a GET request to the documentation.

Reply:
Done